components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -929,7 +929,9 @@ private static String getScopeMetadataQueryParam(Set<String> scopes, String tena
             String scopeMetadata = new Gson().toJson(scopesMetaData);
             return "scopeMetadata=" + URLEncoder.encode(scopeMetadata, UTF_8);
         } catch (Exception e) {
-            log.warn("Error while retrieving scope metadata for scopes: " + scopes, e);
+            if (log.isDebugEnabled()) {
+                log.debug("Error while retrieving scope metadata for scopes: " + scopes, e);
+            }
```

Comment:
Should this be a debug log?

Reply:
As this is not interrupting the flow, decided to add it as a debug log instead of an error or warn log.